src/main/java/com/metamx/emitter/core/ParametrizedUriEmitter.java

```diff
@@ -95,10 +94,23 @@ public void close() throws IOException
   @Override
   public void flush() throws IOException
   {
-    synchronized (emitters) {
-      for (Emitter emitter : emitters.values()) {
-        emitter.flush();
+    Exception thrown = null;
+    for (HttpPostEmitter httpPostEmitter : emitters.values()) {
+      try {
+        httpPostEmitter.flush();
+      }
+      catch (Exception e) {
```

Comment:
This does not handle InterruptedException properly. If the flush was interrupted, the expected behavior is to stop flushing an propagate the error.